File: 4_median_of_two_arrays/median.py
from typing import List


class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        i = 0
        j = 0

        n = len(nums1)
        m = len(nums2)

        res = []

        while i < n and j < m:
            if nums1[i] < nums2[j]:
                res.append(nums1[i])
                i += 1
            else:
                res.append(nums2[j])
                j += 1

        while i < n:
            res.append(nums1[i])
            i += 1

        while j < m:
            res.append(nums2[j])
            j += 1

        q = int(len(res) / 2)
        
        if len(res) % 2 == 0:
            return (res[q] + res[q - 1]) / 2
        return res[q]

File: 4_median_of_two_arrays/Median.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n = nums1.length;
        int m = nums2.length;
        int r = n + m;
        int[] res = new int[r];

        int i = 0;
        int j = 0;
        int k = 0;

        while (i < n && j < m) {
            if (nums1[i] < nums2[j]) {
                res[k] = nums1[i];
                i++;
            } else {
                res[k] = nums2[j];
                j++;
            }
            k++;
        }

        while (i < n) {
            res[k] = nums1[i];
            i++;
            k++;
        }

        while (j < m) {
            res[k] = nums2[j];
            j++;
            k++;
        }

        int q = res.length / 2;

        if (r % 2 == 0) {
            return (res[q] + res[q - 1]) / 2.0;
        }

        return res[q];
    }
}